frontend/src/components/generic/useClipboardEntity.js

```diff
@@ -0,0 +1,87 @@
+import { computed, ref } from 'vue'
+
+export function useClipboardEntity(
+  apiGet,
+  { fetchClipboardEntity, onEntityLoaded, onEntityLoadFailed }
+) {
+  const clipboardPermission = ref('unknown')
+  const clipboardEntity = ref(null)
+  const _clipoardEntityUrl = ref(null)
+  const loading = ref(false)
+
+  const clipboardEntityUrl = computed({
+    get() {
+      return _clipoardEntityUrl.value
+    },
+    set: setClipboardEntityUrl,
+  })
+
+  async function setClipboardEntityUrl(url) {
+    loading.value = true
+    _clipoardEntityUrl.value = url
+
+    return fetchClipboardEntity(url)
+      .then((entityProxy) => {
+        if (entityProxy == null) return Promise.reject('not found')
+        return entityProxy?._meta.load
+      })
+      .then((entity) => {
+        clipboardEntity.value = entity
+        onEntityLoaded(entity)
+        loading.value = false
+      })
+      .catch(() => {
+        clipboardEntity.value = null
+        onEntityLoadFailed(url)
+        loading.value = false
+      })
+  }
+
+  const clipboardAccessDenied = computed(() => {
+    return (
+      clipboardPermission.value === 'unaccessible' ||
+      clipboardPermission.value === 'denied'
+    )
+  })
+
+  const showClipboardPrompt = computed(() => clipboardPermission.value === 'prompt')
+
+  const hasClipboardEntity = computed(() => {
+    return clipboardEntity.value != null && clipboardEntity.value._meta.self != null
+  })
+
+  const attemptLoadingEntityFromClipboard = async () => {
+    clipboardEntity.value = null
+    return navigator.permissions
+      .query({ name: 'clipboard-read' })
+      .then((p) => {
+        clipboardPermission.value = p.state
+        if (p.state !== 'granted') return null
+        return navigator.clipboard.readText()
+      })
+      .then(async (url) => {
+        return setClipboardEntityUrl(url)
+      })
```

Comment:
```suggestion
.then(async (p) => {
   clipboardPermission.value = p.state
    if (p.state === 'granted') {
      const url = await navigator.clipboard.readText()
      await setClipBoardEntity(url)
    }
  })
```
Then we don't have to handle null in setClipBoardEntity.

Reply:
[46770e8](https://github.com/ecamp/ecamp3/pull/7528/commits/46770e89d7dbea2d97c155dd3ad6306fc8763ea2)